src/nodejs/lambda-handlers/workflow-consumer.js

```diff
@@ -146,27 +146,39 @@ async function formSubmittedMethod(eventMessage) {
 }
 
 async function reviewApprovedMethod(eventMessage) {
-  const newEvent = { ...eventMessage, event_type: 'workflow_promote_step_direct' };
-  if (eventMessage.step_name === 'data_accession_request_form_review') {
-    await db.metrics.setAccessionReversion({
-      id: eventMessage.submission_id,
-      status: 'FALSE'
-    });
+  const param = { submission_id: eventMessage.submission_id, step_name: eventMessage.step_name };
+  const stepReview = await db.submission.checkCountStepReviewApproved(param);
+  if (stepReview.unapproved && parseInt(stepReview.unapproved, 10) === 0) {
+    const newEvent = { ...eventMessage, event_type: 'workflow_promote_step_direct' };
+    if (eventMessage.step_name === 'data_accession_request_form_review') {
+      await db.metrics.setAccessionReversion({
+        id: eventMessage.submission_id,
+        status: 'FALSE'
+      });
+    }
+    await promoteStepMethod(eventMessage);
+    await msg.sendEvent(newEvent);
   }
-  await promoteStepMethod(eventMessage);
-  await msg.sendEvent(newEvent);
 }
 
 async function reviewRejectedMethod(eventMessage) {
-  const { submission_id: id, data: { rollback } } = eventMessage;
-  if (eventMessage.step_name === 'data_accession_request_form_review') {
-    await db.metrics.setAccessionReversion({
-      id: eventMessage.submission_id,
-      status: 'TRUE'
-    });
+  const { id, data: { rollback } } = eventMessage;
+  // Did this because of lint error. This line has a length of 104. Maximum allowed is 100
+  const StepName = eventMessage.step_name;
+  const submissionId = eventMessage.submission_id;
+  const userId = eventMessage.user_id;
+  const param = { submission_id: submissionId, step_name: StepName, user_id: userId };
+  const stepReview = await db.submission.checkCountStepReviewRejected(param);
+  if (stepReview.unapproved && parseInt(stepReview.unapproved, 10) === 0) {
```

Comment:
We should not need to get here if someone rejects that is in the step review table